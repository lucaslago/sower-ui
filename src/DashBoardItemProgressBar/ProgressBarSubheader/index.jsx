import React from 'react';
import { Row, Col } from 'react-bootstrap';
import CircularProgress from 'material-ui/CircularProgress';
import Subheader from 'material-ui/Subheader';
import LinearProgress from 'material-ui/LinearProgress';
import SIMULATION_STATUS from '../../utils/simulation_status';

const ProgressBarSubheader = ({ completed, status, totalPositions, remainingPositions }) => {
  if (status === SIMULATION_STATUS.ACTIVE) {
    return (
      <div>
        <Subheader>
          { (totalPositions - remainingPositions) } / { totalPositions } 
        </Subheader>
        <LinearProgress mode="determinate" value={completed} />
      </div>
    )
  }
  return (
    <Row>
      <Col md={2} mdOffset={5} >
        <CircularProgress />
      </Col>
    </Row>
  )
}

ProgressBarSubheader.propTypes = {
  completed: React.PropTypes.number.isRequired,
  status: React.PropTypes.string.isRequired,
  totalPositions: React.PropTypes.number,
  remainingPositions: React.PropTypes.number
};

export default ProgressBarSubheader;
