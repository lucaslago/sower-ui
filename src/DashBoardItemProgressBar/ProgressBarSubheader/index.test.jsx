import ProgressBarSubheader from './index';
import CircularProgress from 'material-ui/CircularProgress';

describe('<ProgressBarSubheader />', () => {
  const props = {
    status: 'active',
    completed: 10,
    totalPositions: 100,
    remainingPositions: 90
  };

  it('should render correctly when status is active', () => {
    const wrapper = shallow(<ProgressBarSubheader {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly when status is inactive', () => {
    const wrapper = shallow(<ProgressBarSubheader {...props} status={'inactive'} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
