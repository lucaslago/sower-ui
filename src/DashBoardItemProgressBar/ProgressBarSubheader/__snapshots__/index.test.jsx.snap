exports[`<ProgressBarSubheader /> should render correctly when status is active 1`] = `
<div>
  <Subheader
    inset={false}>
    10
     / 
    100
  </Subheader>
  <LinearProgress
    max={100}
    min={0}
    mode="determinate"
    value={10} />
</div>
`;

exports[`<ProgressBarSubheader /> should render correctly when status is inactive 1`] = `
<Row
  bsClass="row"
  componentClass="div">
  <Col
    bsClass="col"
    componentClass="div"
    md={2}
    mdOffset={5}>
    <CircularProgress
      max={100}
      min={0}
      mode="indeterminate"
      size={40}
      thickness={3.5}
      value={0} />
  </Col>
</Row>
`;
